sound/soc/sof/sof-priv.h

```diff
@@ -38,6 +38,9 @@ extern int sof_core_debug;
 /* max BARs mmaped devices can use */
 #define SND_SOF_BARS	8
 
+/* max core number */
+#define SND_SOF_CORE_MAX	8
```

Comment:
> I understand 8 is relatively small and we're quite unlikely to ever have more than 8 cores on our DSPs, but maybe still it would be better to be more precise. Our `struct sof_intel_dsp_desc` has a `.cores_num` field, so, we could use it to dynamically allocate an array of per-core status objects.

That's a good point but since the struct sof_intel_dsp_desc.cores_num is Intel specific and the SOF_CORE and multi-core support here is generic, so I take lazy way to just define a maximum count to avoid reading it from some IMX platforms.

Reply:
@lyakh has a point, if the multi-core is supposed to be generic then we are missing something here. Having an 8 here looks hacky.